### PR TITLE
Mon 13113 visual regression tests for the navigation menu using cypress

### PR DESCRIPTION
**Pull request by Noha-ElAbrouki**
visual testing using cypress image snapshot

### CODE REVIEW

File: www/front_src/src/Navigation/Sidebar/index.cypress.spec.tsx [hunk 1]
```diff
@@ -0,0 +1,77 @@
+import { BrowserRouter as Router } from 'react-router-dom';
+
+import { mount } from '../../../../../cypress/support';
+
+import SideBar from './index';
+
+describe('Visual testing for menu: ', () => {
```
Comment (bdauria) [inline www/front_src/src/Navigation/Sidebar/index.cypress.spec.tsx:7]:
```suggestion
describe('Navigation menu visual testing: ', () => {
```

File: www/front_src/src/Navigation/Sidebar/index.cypress.spec.tsx [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+  beforeEach(() => {
+    cy.fixture('dataMenu').then((data) => {
```
Comment (bdauria) [inline www/front_src/src/Navigation/Sidebar/index.cypress.spec.tsx:9]:
```suggestion
    cy.fixture('menuData').then((data) => {
```

File: www/front_src/src/Navigation/Sidebar/index.cypress.spec.tsx [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+      mount(
+        <Router>
+          <SideBar navigationData={data.result} />
+        </Router>,
+      );
+    });
+  });
+  it('Should match previous screenshot "initial menu"', () => {
```
Comment (bdauria) [inline www/front_src/src/Navigation/Sidebar/index.cypress.spec.tsx:17]:
```suggestion
  it('matches the current snapshot "initial menu"', () => {
```

File: www/front_src/src/Navigation/Sidebar/index.cypress.spec.tsx [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+    cy.get("[alt='mini logo']").should('be.visible');
+    cy.get('li').each(($li) => {
+      cy.wrap($li).get('svg').should('be.visible');
+    });
+    cy.matchImageSnapshot();
+  });
+
+  it('menu should expanded after clicking on logo', () => {
```
Comment (bdauria) [inline www/front_src/src/Navigation/Sidebar/index.cypress.spec.tsx:25]:
```suggestion
  it('expands when the logo is clicked', () => {
```

File: www/front_src/src/Navigation/Sidebar/index.cypress.spec.tsx [hunk 5]
```diff
@@ -0,0 +1,77 @@
...
+      } else if (index === 1) {
+        cy.get('@element').contains('Home');
+      } else {
+        cy.get('@element').contains('Configuration');
+      }
+    });
+    cy.matchImageSnapshot();
+  });
+
+  it('collapse should be visible when item of menu is hovered, and it should be colored', () => {
```
Comment (bdauria) [inline www/front_src/src/Navigation/Sidebar/index.cypress.spec.tsx:41]:
```suggestion
  it('displays the direct child items and highlights the item when hovered ', () => {
```

File: www/front_src/src/Navigation/Sidebar/index.cypress.spec.tsx [hunk 6]
```diff
@@ -0,0 +1,77 @@
...
+    cy.get('li').eq(2).trigger('mouseover');
+    cy.get('[data-cy=collapse]').should('be.visible');
+    cy.matchImageSnapshot();
+  });
+
+  it('item of menu should be colored after double click', () => {
```
Comment (bdauria) [inline www/front_src/src/Navigation/Sidebar/index.cypress.spec.tsx:47]:
```suggestion
  it('hilights the menu item when double clicked', () => {
```

File: www/front_src/src/Navigation/Sidebar/index.cypress.spec.tsx [hunk 7]
```diff
@@ -0,0 +1,77 @@
...
+    cy.get('li').eq(1).as('element').trigger('mouseover');
+    cy.get('@element').trigger('dblclick');
+    cy.matchImageSnapshot();
+  });
+
+  it('when item of collapse is clicked , the item and parent item should be colored', () => {
```
Comment (bdauria) [inline www/front_src/src/Navigation/Sidebar/index.cypress.spec.tsx:53]:
```suggestion
  it('higlights the parent item when the item is clicked', () => {
```

**Review by Thebarda — COMMENTED** [review]
Is it possible to change the `snapshots/All Specs` folder to `visual-testing-snapshots` ?

File: www/front_src/src/Navigation/Sidebar/index.cypress.spec.tsx [hunk 8]
```diff
@@ -0,0 +1,85 @@
...
+    cy.matchImageSnapshot();
+  });
+
+  it('hilights the menu item when double clicked', () => {
+    cy.get('li').eq(1).as('element').trigger('mouseover');
+    cy.get('@element').trigger('dblclick');
+    cy.matchImageSnapshot();
+  });
+
+  it('higlights the parent item when the item is clicked', () => {
```
Comment (bdauria) [inline www/front_src/src/Navigation/Sidebar/index.cypress.spec.tsx:61]:
```suggestion
  it('hilights the parent item when the item is clicked', () => {
```

File: www/front_src/src/Navigation/Sidebar/index.cypress.spec.tsx [hunk 9]
```diff
@@ -0,0 +1,85 @@
...
+        <Router>
+          <SideBar navigationData={data.result} />
+        </Router>,
+      );
+    });
+    const { result } = renderHook(() => useAtom(selectedNavigationItemsAtom));
+    act(() => {
+      result.current[1](null);
+    });
+  });
```
Comment (bdauria) [inline www/front_src/src/Navigation/Sidebar/index.cypress.spec.tsx:24]:
```suggestion
  });
  
```

File: www/front_src/src/Navigation/Sidebar/index.cypress.spec.tsx [hunk 10]
```diff
@@ -0,0 +1,85 @@
...
+describe('Navigation menu visual testing: ', () => {
+  beforeEach(() => {
+    cy.fixture('menuData').then((data) => {
+      mount(
+        <Router>
+          <SideBar navigationData={data.result} />
+        </Router>,
+      );
+    });
+    const { result } = renderHook(() => useAtom(selectedNavigationItemsAtom));
```
Comment (bdauria) [inline www/front_src/src/Navigation/Sidebar/index.cypress.spec.tsx:20]:
```suggestion
    const { result } = renderHook(() => useAtom(selectedNavigationItemsAtom));
    
```

File: www/front_src/src/Navigation/Sidebar/index.cypress.spec.tsx [hunk 11]
```diff
@@ -0,0 +1,85 @@
...
+    cy.get('@first_ele_collapse')
+      .find('[data-cy=collapse]')
+      .as('second_collapse')
+      .should('be.visible');
+    cy.get('@second_collapse')
+      .find('ul')
+      .first()
+      .trigger('mouseover')
+      .trigger('click');
+    cy.matchImageSnapshot();
```
Comment (bdauria) [inline www/front_src/src/Navigation/Sidebar/index.cypress.spec.tsx:83]:
```suggestion

    cy.matchImageSnapshot();
```

File: www/front_src/src/Navigation/Sidebar/index.cypress.spec.tsx [hunk 12]
```diff
@@ -0,0 +1,85 @@
...
+    cy.get('li').each(($li, index) => {
+      cy.wrap($li).as('element').get('svg').should('be.visible');
+      if (index === 0) {
+        cy.get('@element').contains('Monitoring');
+      } else if (index === 1) {
+        cy.get('@element').contains('Home');
+      } else {
+        cy.get('@element').contains('Configuration');
+      }
+    });
```
Comment (bdauria) [inline www/front_src/src/Navigation/Sidebar/index.cypress.spec.tsx:45]:
```suggestion

    });
```

File: www/front_src/src/Navigation/Sidebar/index.cypress.spec.tsx [hunk 13]
```diff
@@ -0,0 +1,85 @@
...
+    cy.matchImageSnapshot();
+  });
+
+  it('displays the direct child items and highlights the item when hovered ', () => {
+    cy.get('li').eq(2).trigger('mouseover');
+    cy.get('[data-cy=collapse]').should('be.visible');
+    cy.matchImageSnapshot();
```
Comment (bdauria) [inline www/front_src/src/Navigation/Sidebar/index.cypress.spec.tsx:52]:
```suggestion

    cy.matchImageSnapshot();
```

File: www/front_src/src/Navigation/Sidebar/index.cypress.spec.tsx [hunk 14]
```diff
@@ -0,0 +1,85 @@
...
+    const { result } = renderHook(() => useAtom(selectedNavigationItemsAtom));
+    act(() => {
+      result.current[1](null);
+    });
+  });
+  it('matches the current snapshot "initial menu"', () => {
+    cy.get("[alt='mini logo']").should('be.visible');
+    cy.get('li').each(($li) => {
+      cy.wrap($li).get('svg').should('be.visible');
+    });
```
Comment (bdauria) [inline www/front_src/src/Navigation/Sidebar/index.cypress.spec.tsx:29]:
```suggestion
    });
    
```

**Pull request closed without merging**